R/DataPulls.R

```diff
@@ -229,3 +280,124 @@ getPreferenceScoreDistribution <- function(connectionDetails = NULL,
+#' that uses the follow-up distribution data.
+#' 
+#' If the comparative effectiveness results are stored in a 
+#' relational database server, then please provide connection details 
+#' using DatabaseConnector::createConnectionDetails. If connectionDetails 
+#' parameter is not null, the function will attempt to connect to 
+#' relational database server, and ignore using R dataframe object. 
+#' Note: schema parameter in createConnectionDetails is not used. We use
+#' databaseSchema in function call instead.
+#' 
+#' If the comparative effectiveness results are stored as a R dataframe object,
+#' please make sure connectionDetails is NULL (Default value). The function 
+#' will expect data in R dataframe. This R dataframe object is expected to follow 
+#' camelCase for both object names and variable names.
+#' 
+#'
+#' @template optionalConnectionDetails
+#' @template optionalDatabaseSchema
+#' @template targetId
+#' @template comparatorId
+#' @template databaseId
+#' @template analysisId
+#' @template outcomeId
+#' @param cmFollowUpDist       (optional) A R-dataFrame object with fields named in camelCase containing
+#'                             data from cm_follow_up_dist table in comparative effectiveness data model.
+#' @return                     Tibble with fields Id, Min, P10, P25, Median, P75, P90,Max, group.
+#'                             group is for identifying if the cohort is target vs comparator.
+#'
+#' @examples
+#' \dontrun{
+#' #If your data is in data frames, provide connectionDetails
+#' connectionDetails <- createConnectionDetails(dbms = "postgresql",
+#'                                              user = "joe",
+#'                                              password = "secret",
+#'                                              server = "myserver")
+#' #If connectionDetails is provided, function will attempt to read data from database
+#' example1 <- getFollowUpDaysDistribution(connectionDetails = connectionDetails,
+#'                                         databaseSchema = 'database.schema',
+#'                                         targetId = 100,
+#'                                         comparatorId = 200,
+#'                                         outcomeId = 111,
+#'                                         databaseId = 'HCUP data',
+#'                                         analysisId = 2)
+#' #If no connectionDetails is provided, then function will check for data in dataFrame
+#' example2 <- getFollowUpDaysDistribution(cmFollowUpDist = cmFollowUpDist,
+#'                                         targetId = 100,
+#'                                         comparatorId = 200,
+#'                                         outcomeId = 111,
+#'                                         databaseId = 'HCUP data',
+#'                                         analysisId = 2)
+#' }
+#' @export
+#'
+getFollowUpDistribution <- function(connectionDetails = NULL,
```

Comment:
I think I missed this in my previous reviews, but maybe we should use a `connection` argument instead of a `connectionDetails` argument. Now we will have to reconnect to the database for every function call, which is slow and may be problematic on some platforms like Oracle that only support a limited amount of connections in a time span.